src/api/spec/controllers/person_controller_spec.rb

```diff
@@ -44,13 +61,29 @@
     end
 
     context 'when in LDAP mode' do
+      let!(:old_password) { user.password_digest }
+
       before do
+        request.env['RAW_POST_DATA'] = 'password_has_changed'
         stub_const('CONFIG', CONFIG.merge('ldap_mode' => :on))
-        post :post_userinfo, params: { login: user.login, cmd: 'change_password', format: :xml }
       end
 
-      it 'user is not allowed to change their password' do
-        expect(response.header['X-Opensuse-Errorcode']).to eq('change_password_no_permission')
+      context 'and the user is configured to authorize on the LDAP server' do
+        before do
+          post :post_userinfo, params: { login: user.login, cmd: 'change_password', format: :xml }
+        end
+
+        it { expect(response.header['X-Opensuse-Errorcode']).to eq('change_password_no_permission') }
+        it { expect(old_password).to eq(user.password_digest) }
+      end
+
+      context 'and the user is configured to authorize locally' do
+        before do
+          user.update(ignore_auth_services: true)
+          post :post_userinfo, params: { login: user.login, cmd: 'change_password', format: :xml }
+        end
+
+        it { expect(old_password).to_not eq(user.reload.password_digest) }
```

Comment:
This test would always pass as the password_digest is the has and would never match the old_password. Not sure if it makes sense to compare the old password digest with the new?

Reply:
Not sure what you mean, tbh. The digest is a hash based on the clear text password. So if the password stays the same the hash is the same as well.  Note that there are also tests for the case that the password does not change.

Reply:
Then please call it ``old_password_digest`` because if you read it, it looks like you compare the password with the hash.